### ISSUE TITLE
 stock moves sequence-error created from other modules

### ISSUE THREAD

**Issue by mahmoudmk1** [OCA/stock-logistics-workflow#631]
module: stock_picking_line_sequence
version: 12.0
**Current behavior**
This module does not integrate with other modules such as (sales, purchase )

for example the stock move sequence created from a" sale order" is not correct, instead of (1,2,3,4....etc) you will find (9,999)for all stock move lines?.
also the stock move sequence created from a" purchase order" also is not correct, you will find lines with a sequence step equal to 5 such as (5,10,15,20,25,......etc.)?

for your information I tested the module with a fresh database but the same error.
could you please check that!

**Comment by rousseldenis** [comment]
As I see in code it should work even with sales/purchases...

@LoisRForgeFlow @sergiocorato What do you think ?

**Comment by LoisRForgeFlow** [comment]
cc @AaronHForgeFlow

**Comment by AaronHForgeFlow** [comment]
I agree sequence to '9999' is ugly, but it is defined like that because when updating the record users want the new line to be at the bottom (highest sequence number). When saving the record (the picking) sequences are recomputed so the '9999' will be the proper sequence.

Respect to the sequence number increment 5 by 5 thing, I could not reproduce that. Could you please put here a link to a runbot where that issue is happening. It seems to me other modules are affecting that behavior.

Regards.

**Comment by LoisRForgeFlow** [comment]
Closing as it seems that there must be something else in your set-up. Please reopen if you can share the steps to reproduce in a runbot instance.

**Comment by mahmoudmk1** [comment]
> I agree sequence to '9999' is ugly, but it is defined like that because when updating the record users want the new line to be at the bottom (highest sequence number). When saving the record (the picking) sequences are recomputed so the '9999' will be the proper sequence.
> 
> Respect to the sequence number increment 5 by 5 thing, I could not reproduce that. Could you please put here a link to a runbot where that issue is happening. It seems to me other modules are affecting that behavior.
> 
> Regards.
Do you meant that 9999 sequence is ok !How it comes? please could you check the attachments which they are a screenshots from your runbot.
Best Regards
![Screenshot 2020-03-11 02 15 47](https://user-images.githubusercontent.com/51242024/76370169-a5dbf780-633e-11ea-9e6a-828fc6bc5b84.png)
![Screenshot 2020-03-11 02 16 03](https://user-images.githubusercontent.com/51242024/76370170-a6748e00-633e-11ea-9fc1-463bc742288e.png)

**Comment by LoisRForgeFlow** [comment]
cc @AaronHForgeFlow

**Comment by LoisRForgeFlow** [comment]
@AaronHForgeFlow Confirmed this issue, a fix will be required.

**Comment by AaronHForgeFlow** [comment]
This is fixed here: https://github.com/OCA/stock-logistics-workflow/pull/638
@qzmxs @mahmoudmk1 Could you please review? 

BTW: This fix just set the sequence starting from 1 onward. If you need to keep the same sequence in sale orders and purchase orders in the picking you will need also those modules: https://github.com/OCA/sale-workflow/tree/12.0/sale_order_line_sequence and https://github.com/OCA/purchase-workflow/pull/613

All comments are welcome.

**Comment by qzmxs** [comment]
When the line number 9999 appears, please uninstall stock_picking_line_sequence, after reinstallation, it becomes the serial number 1, 2, 3, 4 ....,
There are no errors in creating outbound orders independently.

**Comment by AaronHForgeFlow** [comment]
Do you mean demo data records?

**Comment by mahmoudmk1** [comment]
The issue solved, thanks for your cooperation. I appreciate your support.
@AaronHForgeFlow @LoisRForgeFlow

**Comment by AaronHForgeFlow** [comment]
Thanks for reporting. Once checks are passed in the PR you can also review it. Regards.

**Comment by qzmxs** [comment]
I don't know how to describe this problem, I expect and suggest that stock documents generated by sales order or purchase order should inherit the serial number

**Comment by AaronHForgeFlow** [comment]
Then you will need also those modules: https://github.com/OCA/sale-workflow/tree/12.0/sale_order_line_sequence and https://github.com/OCA/purchase-workflow/tree/12.0/purchase_order_line_sequence

**Comment by qzmxs** [comment]
purchase_order_line_sequence，This purchasing module has been tested normally, and it is normal to generate an inventory serial number

**Comment by qzmxs** [comment]
The stock number is displayed as 9999, which may be caused by sale_order_line_sequence

**Comment by qzmxs** [comment]
purchase ok
![image](https://user-images.githubusercontent.com/55458897/76524236-abebe880-64a4-11ea-93ee-41e73b52fd51.png)
![image](https://user-images.githubusercontent.com/55458897/76524295-c58d3000-64a4-11ea-9f70-bff22af569fa.png)
sale
![image](https://user-images.githubusercontent.com/55458897/76524363-e786b280-64a4-11ea-8582-6509897dff35.png)
![image](https://user-images.githubusercontent.com/55458897/76524526-2c124e00-64a5-11ea-9516-e8f7bc5c2326.png)

**Comment by AaronHForgeFlow** [comment]
Could not reproduce on runbot. If this is an issue of the purchase_order_line_sequence module please create the issue https://github.com/OCA/purchase-workflow/issues not here.

**Comment by mahmoudmk1** [comment]
@qzmxs  I think I know how to fix the issue you just described with the stock move sequence 9999 created from sale module. I think you should follow these steps:
1-uninstall the old version of  stock_picking_line_sequence and delete the old files from addons path
2-upload the fixed one using this url https://github.com/ForgeFlow/stock-logistics-workflow/tree/12.0-fix-stock_picking_line_sequence/ ,update your apps then install it.
It should works as my case

**Comment by qzmxs** [comment]
@mahmoudmk1
https://github.com/ForgeFlow/stock-logistics-workflow/tree /12.0-fix-stock_picking_line_sequence/
The following problems exist

![image](https://user-images.githubusercontent.com/55458897/76758053-d64ee600-67c3-11ea-99b4-aafa8f5795c6.png)

![image](https://user-images.githubusercontent.com/55458897/76758151-16ae6400-67c4-11ea-8cfe-b89b2ca74c75.png)
![image](https://user-images.githubusercontent.com/55458897/76758168-26c64380-67c4-11ea-976a-854ca10d79fe.png)

**Comment by AaronHForgeFlow** [comment]
thanks for  reporting, if this issue is reproduced in the runbot of the pull request can we followup there?

**Comment by AaronHForgeFlow** [comment]
Could not reproduce this error in the pull request: https://github.com/OCA/stock-logistics-workflow/pull/638

Will check the integration with purchase_order_line_sequence, but as said, that would be an issue of a different module.

**Comment by qzmxs** [comment]
oca's stock_picking_line_sequence has not fixed the previous problem. My problem is based on https://github.com/ForgeFlow/stock-logistics-workflow/tree /12.0-fix-stock_picking_line_sequence/.

Since ForgeFlow's stock_picking_line_sequence cannot submit the issue, I will submit it here so it can be resolved together.

**Comment by qzmxs** [comment]
@AaronHForgeFlow ，The https://github.com/ForgeFlow/stock-logistics-workflow/tree/12.0-fix-stock_picking_line_sequence/stock_picking_line_sequence modified 3 hours ago still did not solve the previous problem. I have uninstalled purchase_order_line_sequence, and I have uninstalled sale_order_line_sequence. Only stock_picking_line_sequence is retained, and the problem persists.

**Comment by qzmxs** [comment]
I set up a new database to test！==

**Comment by qzmxs** [comment]
@AaronHForgeFlow
Brand new database with only purchase_order_line_sequence installed,
![image](https://user-images.githubusercontent.com/55458897/76782113-bda5f680-67ea-11ea-961d-e1592af238f8.png)
![image](https://user-images.githubusercontent.com/55458897/76782244-f2b24900-67ea-11ea-8a27-2f6a1a7a760b.png)
![image](https://user-images.githubusercontent.com/55458897/76782304-107fae00-67eb-11ea-8845-a62287095047.png)
![image](https://user-images.githubusercontent.com/55458897/76782367-255c4180-67eb-11ea-8ad2-083451215916.png)
![image](https://user-images.githubusercontent.com/55458897/76782456-4755c400-67eb-11ea-8e6d-a30d82ad0d34.png)

**Comment by AaronHForgeFlow** [comment]
I am sorry but could not reproduce in local. Could you reproduce the same on runbot.

In case you are trying to reproduce in local please ensure you fetched the latest code and there is no external module affecting the behavior. Regards.

**Comment by AaronHForgeFlow** [comment]
I could reproduce now. I will fix that. Thanks for letting me know.

**Comment by AaronHForgeFlow** [comment]
@qzmxs I think now the issue is solved. Could you pleas review in the PR? Thanks

**Comment by qzmxs** [comment]
The latest version of ForgeFlows（tock_picking_line_sequence）
The latest version of OCA（sale_order_line_sequence）
The latest version of OCA（purchase_order_line_sequence
The previous issue was resolved after installation, but the sales order generation stock list duplicated the historical 9999 issue

Created separately stock out order
![image](https://user-images.githubusercontent.com/55458897/77081564-cd654b00-6a35-11ea-95e6-e63ad0276378.png)

Other normal

**Comment by qzmxs** [comment]
Error 1
![image](https://user-images.githubusercontent.com/55458897/77093000-5b94fd80-6a45-11ea-9526-a20b00bf5583.png)
![image](https://user-images.githubusercontent.com/55458897/77093092-78c9cc00-6a45-11ea-8954-7185d644d27d.png)
Error 2
![image](https://user-images.githubusercontent.com/55458897/77093290-c2b2b200-6a45-11ea-869e-080955766ec2.png)
![image](https://user-images.githubusercontent.com/55458897/77093463-f42b7d80-6a45-11ea-8e7f-6462b42f50da.png)

**Comment by mahmoudmk1** [comment]
@AaronHForgeFlow @qzmxs  I tested the latest version for modules 
-"sale_order_line_sequence-V12.0.1.0.0" as in https://github.com/OCA/sale-workflow/tree/12.0/sale_order_line_sequence 
-"purchase_order_line_sequence-V12.0.1.0.1" as in https://github.com/OCA/purchase-workflow/tree/12.0/purchase_order_line_sequence
WITH
-"stock_picking_line_sequence-V12.0.1.1.1" AS IN https://github.com/ForgeFlow/stock-logistics-workflow/tree/12.0-fix-stock_picking_line_sequence/stock_picking_line_sequence 
SO
 but a new issue has found that picking sequence lines for any new created stock move are 9999,9999,9999,..... during creating and it terns into 1,2,3,......only after save .and also for stock picking created from sales order
. exactly as @qzmxs comment in Mar 19 !!!

**Comment by AaronHForgeFlow** [comment]
@mahmoudmk1 having default sequence 9999 is necessary so the new lines are at the end. When saving the lines are sorted correctly. IMHO we should follow up in the pull request (https://github.com/OCA/stock-logistics-workflow/pull/638) and not here. I need reviewers on that pull requests otherwise it will never be merged.
